taskvine/src/worker/vine_cache.c

```diff
@@ -198,6 +199,52 @@ static int do_command( struct vine_cache *c, const char *command, const char *ca
 	return result;
 }
 
+/*
+Transfer a single input file from a worker url to a local file name. 
+
+*/
+static int do_worker_transfer( struct vine_cache *c, const char *source_url, const char *cache_path, char **error_message)
+{	
+	int port_num;
+	char *addr, *port, *path;
+	char source_mutable[4096];
+	
+	strcpy(source_mutable, source_url);
+	
+	// expect the form: worker://addr:port/path/to/file
+	strtok(source_mutable, "/:");
+	addr = strtok(NULL, "/:");
+	port = strtok(NULL, "/:");
+	path = strtok(NULL, "");
+
+	if(path == NULL)
+	{ 
+		debug(D_VINE, "Improperly formed worker URL: %s", source_url);
+		return 0;
+	}
+
+	port_num = atoi(port);
+	int stoptime = time(0) + 15;
+	struct link *worker_link = link_connect(addr, port_num, stoptime);
+
+	if(worker_link == NULL)
+	{
+		debug(D_VINE, "Could not connect with worker at: %s", source_url);
+	}
+
+	if(!vine_transfer_get_any(worker_link, c, path, time(0) + 120))
+	{
+		return 0;
+	}
+
+	// rename file to our expected cache name (probably not the best way to do this)
```

Comment:
The program `mv` is basically a big wrapper around the system call `rename`.
You can just call `rename` directly here and save some trouble.